.github/workflows/test.yml

```diff
@@ -11,58 +11,57 @@ on:
       - Cargo.toml
 
 jobs:
-  test:
+  timezones_linux:
     strategy:
       matrix:
-        os: [ubuntu-20.04, ubuntu-latest, macos-latest, windows-latest]
-        rust_version: [stable]
-        exhaustive_tz: [onetz]
-        check_combinatoric: [no_combinatoric]
-        include:
-          # check all tzs on most-recent OS's
-          - os: ubuntu-latest
-            rust_version: stable
-            exhaustive_tz: all_tzs
-          - os: windows-latest
-            rust_version: stable
-            exhaustive_tz: all_tzs
-          - os: macos-latest
-            rust_version: stable
-            exhaustive_tz: all_tzs
-          # compilation check
-          - os: ubuntu-latest
-            rust_version: stable
-            check_combinatoric: "combinatoric"
-          # test other rust versions
-          - os: ubuntu-latest
-            rust_version: beta
-          - os: ubuntu-latest
-            rust_version: nightly
-          - os: ubuntu-20.04
-            rust_version: 1.38.0
-          - os: macos-latest
-            rust_version: 1.38.0
-          - os: windows-latest
-            rust_version: 1.38.0
-
+        os: [ubuntu-latest]
+        tz: ["ACST-9:30", "EST4", "UTC0", "Asia/Katmandu"]
     runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v1
+      - run: cargo test --all-features --color=always -- --color=always
 
+  timezones_other:
+    strategy:
+      matrix:
+        os: [macos-latest, windows-latest]
+        tz: ["ACST-9:30", "EST4", "UTC0", "Asia/Katmandu"]
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v1
+      - run: cargo test --lib --all-features --color=always -- --color=always
+      - run: cargo test --doc --all-features --color=always -- --color=always
 
-      - name: Install rust
-        uses: actions-rs/toolchain@v1
+  rust_versions:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        rust_version: ["1.38.0", "stable", "beta", "nightly"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dtolnay/rust-toolchain@master
         with:
           toolchain: ${{ matrix.rust_version }}
-          override: true
+      - uses: Swatinem/rust-cache@v1
+      - run: cargo test --lib --features serde --color=always -- --color=always
```

Comment:
Why does this separate `--lib` and `--doc`? Why do we include serde here, but not other features? I think we should ideally test `--all-features` across the Rust versions, or do we have features that don't work on 1.38?

Reply:
I've separated `--lib` and `--doc` to avoid the integration tests, specifically the long-running test against the `date` command (that is run in some other specific cases) - however it would probably be cleaner to use `--all-features` here.

It looks like the `rkyv` feature doesn't work on `1.38`, but this can be solved by not using the default features.